packages/icon/package.py

```diff
@@ -0,0 +1,247 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os, subprocess
+from spack import *
+
+
+class Icon(AutotoolsPackage):
+    """The ICON modelling framework is a joint project between the
+    German Weather Service and the
+    Max Planck Institute for Meteorology for
+    developing a unified next-generation global numerical weather prediction and
+    climate modelling system. The ICON model has been introduced into DWD's
+    operational forecast system in January 2015."""
+
+    homepage = 'https://software.ecmwf.int/wiki/display/GRIB/Home'
+    url = 'https://gitlab.dkrz.de/icon/icon-cscs/-/archive/mc10_for_icon-2.6.x-rc/icon-cscs-mc10_for_icon-2.6.x-rc.tar.gz'
+    git = 'git@gitlab.dkrz.de:icon/icon-cscs.git'
+
+    maintainers = ['egermann']
+
+    version('master', branch='master', submodules=True)
+    version('ham', git='git@git.iac.ethz.ch:hammoz/icon-hammoz.git', branch='hammoz/gpu/master', submodules=True)
+    version('2.6.x-rc', commit='040de650', submodules=True)
+    version('2.0.17', commit='39ed04ad', submodules=True)
+
+    depends_on('m4')
+    depends_on('autoconf%gcc')
+    depends_on('automake%gcc')
+    depends_on('libtool%gcc')
+    depends_on('cmake%gcc')
+    depends_on('libxml2@2.9.7%gcc', type=('build', 'link', 'run'))
+    depends_on('serialbox@2.6.0', when='+serialize', type=('build', 'link', 'run'))
+    depends_on('eccodes@2.18.0 +build_shared_libs', when='+eccodes', type=('build', 'link', 'run'))
+    depends_on('hdf5', type=('build', 'link', 'run'))
+    depends_on('claw@2.0.1', when='+claw', type=('build', 'link', 'run'))
+    depends_on('netcdf-c ~mpi', type=('build', 'link', 'run'))
+    depends_on('mpicuda', when='icon_target=gpu', type=('build', 'link', 'run'))
+    depends_on('ompt-openmp', when='%pgi', type=('build', 'link', 'run'))
+    depends_on('mpi', when='icon_target=cpu', type=('build', 'link', 'run'))
```

Comment:
This looks suspicious. We also need MPI for the GPU code. This `when` would be more appropriate for the openmp library one line above.

Reply:
The mpicuda is there when we need the GPU code, on Daint it is the same as mpi but not on tsa thus both dependencies. Mpi is however there for both target no worries